src/__tests__/unit/lib/sci/index.test.ts

```diff
@@ -1,472 +1,340 @@
+            'embodied-carbon': 0.05,
+            'functional-unit': 'requests',
+            'functional-unit-time': '1 day',
+            carbon: 0.0025,
+            duration: 100,
+            sci: 216,
+          },
+          {
+            timestamp: '2021-01-01T00:00:00Z',
+            'operational-carbon': 0.002,
+            'embodied-carbon': 0.0005,
+            'functional-unit': 'requests',
+            'functional-unit-time': '1 day',
+            carbon: 0.00125,
+            duration: 2,
+            sci: 108,
+          },
+        ]);
+      });
+
+      it('throws an excetion on missing functional unit data.', async () => {
+        const inputs = [
+          {
+            timestamp: '2021-01-01T00:00:00Z',
+            'operational-carbon': 0.002,
+            'embodied-carbon': 0.0005,
+            'functional-unit': 'requests',
+            duration: 1,
+          },
+        ];
+        expect.assertions(1);
+
+        try {
+          await sciModel.execute(inputs);
+        } catch (error) {
+          expect(error).toBeInstanceOf(InputValidationError);
+        }
+      });
+
+      it('throws exception on invalid functional unit data.', async () => {
+        const inputs = [
+          {
+            timestamp: '2021-01-01T00:00:00Z',
+            'operational-carbon': 0.002,
+            'embodied-carbon': 0.0005,
+            'functional-unit': 'requests',
+            'functional-unit-time': 'bad-data',
+            duration: 1,
+          },
+        ];
+
+        expect.assertions(1);
+
+        try {
+          await sciModel.execute(inputs);
+        } catch (error) {
+          expect(error).toBeInstanceOf(InputValidationError);
+        }
+      });
+
+      it('throws exception on negative time value.', async () => {
+        const inputs = [
+          {
+            timestamp: '2021-01-01T00:00:00Z',
+            'operational-carbon': 0.002,
+            'embodied-carbon': 0.0005,
+            'functional-unit': 'requests',
+            'functional-unit-time': '-1 hour',
+            duration: 1,
+          },
+        ];
+
+        expect.assertions(1);
+
+        try {
+          await sciModel.execute(inputs);
+        } catch (error) {
+          expect(error).toBeInstanceOf(InputValidationError);
+        }
+      });
+
+      it('throws exception on invalid time unit.', async () => {
+        const inputs = [
+          {
+            timestamp: '2021-01-01T00:00:00Z',
+            'operational-carbon': 0.002,
+            'embodied-carbon': 0.0005,
+            'functional-unit': 'requests',
+            'functional-unit-time': '1 badData',
+            duration: 1,
+          },
+        ];
+        expect.assertions(1);
+
+        try {
+          await sciModel.execute(inputs);
+        } catch (error) {
+          expect(error).toBeInstanceOf(InputValidationError);
+        }
+      });
+
+      it('accepts underscore separated values in functional-unit-time.', async () => {
```

Comment:
terms in test names can be introduced in ` `

Reply:
done